crates/utils/src/utils/validation.rs

```diff
@@ -95,13 +95,41 @@ pub fn is_valid_actor_name(name: &str, actor_name_max_length: usize) -> LemmyRes
   }
 }
 
+fn has_3_permitted_display_chars(name: &str) -> bool {
+    let mut num_non_fdc: i8 = 0;
+    for c in name.chars() {
+        if !FORBIDDEN_DISPLAY_CHARS.contains(&c) {
+            num_non_fdc += 1;
+            if num_non_fdc >= 3 {
+                break;
+            }
+        }
+    }
+    if num_non_fdc >= 3 {
+        return true;
+    }
+    return false;
+}
+
+fn invisibly_starts_with_at(name: &str) -> bool {
```

Comment:
Why is this function necessary tho now? The check before was probably fine: `!name.starts_with('@')` . You shouldn't need to loop through every char to find the first character.

Reply:
I assumed from `// Make sure zero-space with an @ doesn't work` that one wasn't supposed to be able to make it look like the display name starts with `@`. It isn't necessary if you are ok with `[FORBIDDEN]@` being accepted.

Reply:
They're separate checks. I'd also adjust your function above to also make sure that it doesn't start with a forbidden char.

Some cases to make sure they fail:

`@anything`

`[ForbiddenChar]anything`

Reply:
What is separate checks?

I could also add !name.starts_with(F...) to is_valid_display_name?

Reply:
I just mean keep the `!name.starts_with(@)` check that you removed.